src/main/flight/pid.c

```diff
@@ -515,20 +515,24 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
             previousRateError[axis] = rD;
 
             // if crash recovery is on and accelerometer enabled then check for a crash
-            if (pidProfile->crash_recovery && ARMING_FLAG(ARMED)) {
-                if (motorMixRange >= 1.0f && inCrashRecoveryMode == false
-                        && ABS(delta) > crashDtermThreshold
-                        && ABS(errorRate) > crashGyroThreshold
-                        && ABS(getSetpointRate(axis)) < crashSetpointThreshold) {
-                    inCrashRecoveryMode = true;
-                    crashDetectedAtUs = currentTimeUs;
-                }
-                if (cmpTimeUs(currentTimeUs, crashDetectedAtUs) < crashTimeDelayUs && (ABS(errorRate) < crashGyroThreshold
-                    || ABS(getSetpointRate(axis)) > crashSetpointThreshold)) {
+            if (pidProfile->crash_recovery) {
+                if (ARMING_FLAG(ARMED)) {
+                    if (motorMixRange >= 1.0f && inCrashRecoveryMode == false
+                       && ABS(delta) > crashDtermThreshold
+                       && ABS(errorRate) > crashGyroThreshold
+                       && ABS(getSetpointRate(axis)) < crashSetpointThreshold) {
+                        inCrashRecoveryMode = true;
+                        crashDetectedAtUs = currentTimeUs;
+                    }
+                    if (cmpTimeUs(currentTimeUs, crashDetectedAtUs) < crashTimeDelayUs && (ABS(errorRate) < crashGyroThreshold
+                       || ABS(getSetpointRate(axis)) > crashSetpointThreshold)) {
+                        inCrashRecoveryMode = false;
+                    }
+                } else {
                     inCrashRecoveryMode = false;
+                    BEEP_OFF;
```

Comment:
this will disable beeper whit each pid loop cycle.
thats wasting cycles and has side effects with other components using the beeper.

Reply:
This is already fixed ;) look my latest PR.

Reply:
ahh. forget it.. allready "fixed" with another merge..
i wonder why such a change get merged that fast with an alternative pr opened and an ongoing dissusion.
i would have waited for feedback from the original author aka. @martinbudden

Reply:
i think because its pretty simple.. idk who decide what in here ;)

Reply:
with 4 pull request it doesnt look pretty simple. sometimes is better to wait and check.
this function already has a big complexity and then such a fast merged just wonders me.

Reply:
As i said, this is not my choice anyway. I just try to help when i can ;)

Reply:
I know that it isnt your choice ;) and I appreciate your help. 

Those message are for the Betaflight Group Members. Changes on important files like pid.c I (personal opion) would not merged that fast. Especial if they are from people that are quite new with the code( nothing against your person) also this code is way to complex to review it in just an hour. It just looks like some lines, but as we found out in @Vidalcris  first pull request it is quite confusing and was not correct plenty of times. 

And with 3.1 we spended plenty of time reducing branches in the pidController and now we are adding one if after the other.
For Blackbox disarm is handled in the disarm function and here it is handled in the loop.

Open review from the original author and an open pull request.
Those are way to much points open for me to simple merge a pr, but this is just my personal opinion. Team Member here are open to do what ever they want.

Reply:
@wind0r: There is a practice that some Betaflight developers are adhering to: 'Make it work, then make it work well'. Maybe you should try using that for your work, and especially when judging the work of outside contributors?

Reply:
Expecial because it because it is a safety hazard and a bugfix patch I prefer one clean pull request.
A clean pull request may take a little more time but in my opinion this bugfix got way to big, because it is complex code and of fast actions.

i am totaly for "make it work, then make it work well" but for me that doesnt count for bug fixes.

I think all is good and as i said above I apperciate @Vidalcris work and your effort.
else i wouldnt have helped with the inital pull request and agreed on talking on ts / skype.

I think i am just to much for a clean git commit history for betaflight ;) similar like dominc to baseflight :D

Reply:
Agreed that this could have been done better. This was something I had on my to-do list to investigate and then fix, and I was delighted to see somebody else pick it up, and happy to help with that even if it meant that there was some friction (and hopefully learning experience) along the way.

Re clean commit history / clean code, I think for any open source project with community input, a balance has to be struck there between cleanliness and openness, as not to discourage contributions from the community - at least this approach seems to have worked in Betaflight's favour in the past. ;-)